src/uucore/src/lib/mods/backup_control.rs

```diff
@@ -17,15 +99,151 @@ the VERSION_CONTROL environment variable.  Here are the values:
+    /// Argument 'existing', 'nil'
     ExistingBackup,
 }
 
-pub fn determine_backup_suffix(supplied_suffix: Option<&str>) -> String {
+/// Backup error types.
+///
+/// Errors are currently raised by [`determine_backup_mode`] only. All errors
+/// are implemented as [`UCustomError`] for uniform handling across utilities.
+#[derive(Debug, Eq, PartialEq)]
+pub enum BackupError {
+    /// An invalid argument (e.g. 'foo') was given as backup type. First
+    /// parameter is the argument, second is the arguments origin (CLI or
+    /// ENV-var)
+    InvalidArgument(String, String),
+    /// An ambiguous argument (e.g. 'n') was given as backup type. First
+    /// parameter is the argument, second is the arguments origin (CLI or
+    /// ENV-var)
+    AmbiguousArgument(String, String),
+    /// Currently unused
+    BackupImpossible(),
+    // BackupFailed(PathBuf, PathBuf, std::io::Error),
+}
+
+impl UCustomError for BackupError {
+    fn code(&self) -> i32 {
+        match self {
+            BackupError::BackupImpossible() => 2,
+            _ => 1,
+        }
+    }
+
+    fn usage(&self) -> bool {
+        // Suggested by clippy.
+        matches!(
+            self,
+            BackupError::InvalidArgument(_, _) | BackupError::AmbiguousArgument(_, _)
+        )
+    }
+}
+
+impl Error for BackupError {}
+
+impl Display for BackupError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        use BackupError as BE;
+        match self {
+            BE::InvalidArgument(arg, origin) => write!(
+                f,
+                "invalid argument ‘{}’ for ‘{}’\n{}",
+                arg, origin, VALID_ARGS_HELP
+            ),
+            BE::AmbiguousArgument(arg, origin) => write!(
+                f,
+                "ambiguous argument ‘{}’ for ‘{}’\n{}",
+                arg, origin, VALID_ARGS_HELP
+            ),
+            BE::BackupImpossible() => write!(f, "cannot create backup"),
+            // BE::BackupFailed(from, to, e) => write!(
+            //     f,
+            //     "failed to backup '{}' to '{}': {}",
+            //     from.display(),
+            //     to.display(),
+            //     e
+            // )
+            // This is for when #2500 lands in master
+            // Display::fmt(
+            //     &uio_error!(e, "failed to backup '{}' to '{}'", from.display(), to.display()),
+            //     f
+            // ),
+        }
+    }
+}
+
+/// Arguments for backup-related functionality.
+///
+/// Rather than implementing the `clap`-Arguments for every utility, it is
+/// recommended to include the `clap` arguments via the functions provided here.
+/// This way the backup-specific arguments are handled uniformly across
+/// utilities and can be maintained in one central place.
+pub mod arguments {
+    extern crate clap;
+
+    pub static OPT_BACKUP: &str = "backup";
+    pub static OPT_BACKUP_NO_ARG: &str = "b";
+    pub static OPT_SUFFIX: &str = "suffix";
```

Comment:
Well it depends on your definition of "hard to debug" really. ;-)

While implementing I accidentally inserted the 'suffix' arg twice into install, which made install fail *every single test*, because `clap` would panic upon processing the arguments.

And furthermore there are only 4 utils (that I know of... ?) that implement `backup` in the first place. But I guess you're right, it wouldn't hurt to rename these. Would if be okay then if I "hardcode" the options names ("suffix", "backup", "b") into the code? Like so:

```rust
    pub static OPT_BACKUP_NO_ARG: &str = "backupopt_b";

    /// '-b' argument
    pub fn backup_no_args() -> clap::Arg<'static, 'static> {
        clap::Arg::with_name(OPT_BACKUP_NO_ARG)
            .short("b")
            .help("like --backup but does not accept an argument")
    }
```

Reply:
Ah that is clearer than what I thought, indeed. The suggestion looks good!